text/0060-refashion-updates-apis.md

```diff
@@ -0,0 +1,416 @@
+                "words",
+                "proximity",
+                "attribute",
+                "wordsPosition",
+                "exactness"
+            ]
+        },
+        "duration": null,
+        "enqueuedAt": "2021-08-10T14:29:17.000000Z",
+        "startedAt": null,
+        "finishedAt": null
+}
+```
+
+e.g. A fully qualified `task` object in `processing` state.
+
+```json
+{
+        "uid": 0,
+        "indexUid": "movies",
+        "status": "processing",
+        "type": "settingsUpdate",
+        "details": {
+            "rankingRules": [
+                "typo",
+                "desc(ranking)",
+                "words",
+                "proximity",
+                "attribute",
+                "wordsPosition",
+                "exactness"
+            ]
+        },
+        "duration": null,
+        "enqueuedAt": "2021-08-10T14:29:17.000000Z",
+        "startedAt": "2021-08-10T14:29:18.000000Z",
+        "finishedAt": null,
+}
+```
+
+e.g. A fully qualified `task` object in `succeeded` state.
+
+```json
+{
+        "uid": 0,
+        "indexUid": "movies",
+        "status": "succeeded",
+        "type": "settingsUpdate",
+        "details": {
+            "rankingRules": [
+                "typo",
+                "desc(ranking)",
+                "words",
+                "proximity",
+                "attribute",
+                "wordsPosition",
+                "exactness"
+            ]
+        },
+        "duration": 1.0,
+        "enqueuedAt": "2021-08-10T14:29:17.000000Z",
+        "startedAt": "2021-08-10T14:29:18.000000Z",
+        "finishedAt": "2021-08-10T14:29:19.000000Z"
+}
+```
+
+e.g. A fully qualified `task` object in `failed` state.
+
+```json
+{
+        "uid": 0,
+        "indexUid": "movies",
+        "status": "failed",
+        "type": "settingsUpdate",
+        "details": {
+            "rankingRules": [
+                "typo",
+                "desc(ranking)",
+                "words",
+                "proximity",
+                "attribute",
+                "wordsPosition",
+                "exactness"
+            ]
+        },
+        "error": {
+            "message": "invalid criterion wordsPosition",
+            "code": "internal",
+            "type": "internal_error",
+            "link": "https://docs.meilisearch.com/errors#internal",
+        },
+        "duration": 1.0,
+        "enqueuedAt": "2021-08-10T14:29:17.000000Z",
+        "startedAt": "2021-08-10T14:29:18.000000Z",
+        "finishedAt": "2021-08-10T14:29:19.000000Z",
+}
+```
+
+e.g. A summarized `task` object as a response for a `202 Accepted` HTTP code.
+
+```json
+{
+    "uid": 0,
+    "indexUid": "movies",
+    "status": "enqueued",
+    "enqueuedAt": "2021-08-11T09:25:53.000000Z"
+}
+```
+
+---
+
+#### 5. APIs endpoints
+
+**Get all tasks** | `GET` - `/tasks`
+
+##### Goals
+
+Allows users to list tasks globally regardless of the indexes involved. Particularly useful to visualize all the tasks.
```

Comment:
So, this gives a full list of all task/update operations ever requested across an instance's whole history, right? Do we have any data on how big this can actually get? Could it become a problem if an instance has been running for a **very** long time and the resulting `task` response is huge?

Reply:
Additionally: is the record of tasks/updates maintained if an instance is rebooted? Like, if the instance (or the machine in which it is running) crashes, will I be able to access previous updates/tasks?

Reply:
"So, this gives a full list of all task/update operations ever requested across an instance's whole history, right?" Absolutely. 

This can indeed represent a large volume of data in a specific case. Having several indexes in which entries are made regularly. On the product side, we planned to add pagination (It should probably be keyset pagination based on consistency and performance) and a default limit very soon.

So yes it could become a problem on the specific case mentioned above while we can't paginate it yet. The advantage is that the user will always have access to the list of tasks by index, it is not removed.

I believe that if the crash occurs during a transaction, the task is lost.  @MarinPostma Is it true?
If the machine reboots to a stable state, the previous tasks are stored and therefore accessible.

Reply:
My follow-up question was going to be about pagination, so thanks for clearing that up!

As for my second question, my apologies, I realised I actually already knew the reply: if an instance is interrupted during an update/task, the update/task will be marked as enqueued, as stated in the [async update guide](https://docs.meilisearch.com/learn/advanced/asynchronous_updates.html#terminate-meilisearch-while-a-task-is-being-processed).